app/allocation/controllers/create/save.js

```diff
@@ -7,13 +7,29 @@ const CreateAllocationBaseController = require('./base')
 class SaveController extends CreateAllocationBaseController {
   async saveValues(req, res, next) {
     try {
-      const data = omit(req.sessionModel.toJSON(), [
+      const prisonerCategoryLabels = [
+        'prisoner_female_category',
+        'prisoner_male_category',
+        'prisoner_youth_female_category',
+        'prisoner_youth_male_category',
+      ]
+      const sessionModel = req.sessionModel.toJSON()
+      const data = omit(sessionModel, [
         'csrf-secret',
         'errors',
         'errorValues',
+        ...prisonerCategoryLabels,
       ])
+
+      const prisonerCategoryLabel = prisonerCategoryLabels.find(
+        it => (sessionModel[it] || '').length > 0
+      )
+
       const allocation = await allocationService.create({
         ...data,
+        ...(prisonerCategoryLabel
+          ? { prisoner_category: sessionModel[prisonerCategoryLabel] }
```

Comment:
I wonder if you can simplify this part of the form journey and handle this on the form step where these fields occur. You could use the `saveValues` method to set the value of `prisoner_category` to the correct value in the session and keep this save method much more simple.

Reply:
Here's [an example](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/master/app/move/controllers/create/court-hearings.js#L78-L108) of where we've done that previously.

Reply:
Fixed.